src/Team-Services-Bot.Api/Dialogs/ConnectDialog.cs

```diff
@@ -65,8 +64,15 @@ public Task StartAsync(IDialogContext context)
 
         private static string GeneratePin()
         {
-            var generator = new Random();
-            return generator.Next(Min, Max).ToString("0000", CultureInfo.InvariantCulture);
+            using (var generator = new RNGCryptoServiceProvider())
+            {
+                var data = new byte[4];
+
+                generator.GetBytes(data);
+                var value = BitConverter.ToUInt16(data, 0);
```

Comment:
Why get 4 bytes if you are only going to convert 2 of them? Also 16 bits gives you only 65536 values out of a possible 99999. Perhaps you could use all 4 bytes and just take the least significant 5 digits of the number, so convert to a uint32 and then modulo 100,000.